edx_rest_framework_extensions/auth/jwt/decoder.py

```diff
@@ -173,18 +173,73 @@ def _set_filters(token):
+    #   continue on to the original code path and try all keys (including symmetric)
+    #   and add monitoring to let us know. This is meant to be temporary, until we
+    #   can fully retire code paths for symmetric keys, as part of
+    #   DEPR: Symmetric JWTs: https://github.com/openedx/public-engineering/issues/83
+
+    # Use add_symmetric_keys=False to only include asymmetric keys at first
+    key_set = _get_signing_jwk_key_set(jwt_issuer, add_symmetric_keys=False)
+    # .. custom_attribute_name: jwt_auth_verify_asymmetric_keys_count
+    # .. custom_attribute_description: Number of JWT verification keys in use for this
+    #   verification. Should be same as number of asymmetric public keys. This is
+    #   intended to aid in key rotations; once the average count stabilizes at a
+    #   higher number after adding a public key, it should be safe to change the secret key.
+    set_custom_attribute('jwt_auth_verify_asymmetric_keys_count', len(key_set))
+
+    try:
+        _ = JWS().verify_compact(token, key_set)
+        # .. custom_attribute_name: jwt_auth_asymmetric_verified
+        # .. custom_attribute_description: The count of JWTs successfully verified
+        #   using an asymmetric key.
+        set_custom_attribute('jwt_auth_asymmetric_verified', True)
+        return
+    except Exception:  # pylint: disable=broad-except
+        # Continue to the old code path of trying all keys
+        pass
+
+    # The following is the original code that includes both the symmetric and asymmetric keys
+    #   as requested with the decode_symmetric_token argument. Note that the check against
+    #   the asymmetric keys here is redundant and unnecessary, but this code is temporary and
+    #   will be simplified once symmetric keys have been fully retired.
+
     key_set = _get_signing_jwk_key_set(jwt_issuer, add_symmetric_keys=decode_symmetric_token)
-    # .. custom_attribute_name: jwt_auth_verify_keys_count
+    # .. custom_attribute_name: jwt_auth_verify_all_keys_count
     # .. custom_attribute_description: Number of JWT verification keys in use for this
     #   verification. Should be same as number of asymmetric public keys, plus one if
     #   a symmetric key secret is set. This is intended to aid in key rotations; once
     #   the average count stabilizes at a higher number after adding a public key, it
     #   should be safe to change the secret key.
-    set_custom_attribute('jwt_auth_verify_keys_count', len(key_set))
+    set_custom_attribute('jwt_auth_verify_all_keys_count', len(key_set))
 
     try:
         _ = JWS().verify_compact(token, key_set)
+        # .. custom_attribute_name: jwt_auth_symmetric_verified
+        # .. custom_attribute_description: The count of JWTs successfully verified
+        #   using a symmetric key.
+        # Note: A separate custom attribute is used in case there are different JWTs decoded
+        #   in the same request.
```

Comment:
I'm having trouble understanding this comment. What other attribute is this being contrasted to?

Reply:
I tried a longer version. You can let me know if it is more clear.